packages/registry/docs/conceptual-guides/add-on-loader.md

```diff
@@ -0,0 +1,97 @@
+---
+myst:
+  html_meta:
+    "description": "An explanation of the add-ons configuration loader in @plone/registry"
+    "property=og:description": "An explanation of the add-ons configuration loader in @plone/registry"
+    "property=og:title": "Add-ons configuration loader"
+    "keywords": "@plone/registry, registry, add-ons, loader"
+---
+
+# Add-ons configuration loader
+
+The `@plone/registry` compatible add-ons are able to load configuration into the configuration registry.
+This is achieved by using the `main` key entry point specified in the add-on {file}`package.json`.
+This should be a JavaScript or TypeScript file, eg. {file}`index.ts` or {file}`index.js` placed somewhere (generally in the root) of your add-on.
+
+```json
+{
+  "main": "index.ts",
+}
+```
+
+This file should contain a default export with a function with this signature:
+
+```ts
+import type { ConfigType } from '@plone/registry';
+
+export default function loadConfig(config: ConfigType) {
+  // You can mutate the configuration object in here
+  return config;
+}
+```
+
+`@plone/registry` has a helper utility `createAddonsLoader` for generating an add-ons loader file that contains the code needed to load the add-ons configuration of all the registered add-ons, keeping the order in which they were defined.
+
+This loader is a JavaScript file and it is placed in the root of your application.
+By default, it's called {file}`.registry.loader.js`.
+
+```{important}
+This file is generated and maintained by `@plone/registry`.
+You should not modify it and add your own styles in here.
+It will be overwriten in the next bundler run.
+```
+
+The add-ons loader generator is meant to be run before bundling your app or by the bundler itself when it runs.
+The `@plone/registry` Vite plugin generates this file so the framework can load it on app bootstrap time.
+
+```js
+  const projectRootPath = path.resolve('.');
+  const { registry, shadowAliases } = AddonRegistry.init(projectRootPath);
+
+  createAddonsLoader(
+    registry.getAddonDependencies(),
+    registry.getAddons(),
+    { tempInProject: true },
+  );
+```
+
+This will create {file}`.registry.loader.js` in the root of your app.
+Afterwards, you have to make sure that your app loads the code on the app bootstrap, as early as possible in your client (and server) code and as a module side-effect.
```

Comment:
This sentence should be a new paragraph, so it doesn't merge with the previous code example and sentence that refers to it.

```suggestion

Afterwards, configure your app to load the code during the app bootstrap, as early as possible in both your client and server code, and as a module side-effect, as shown in the following example.
```